### PR TITLE
[OMN-189] : Settings View - Emails

### PR DESCRIPTION
**Pull request by gitstart-omnivore**
## Spec Checklist
- [ ] Are Api endpoints ready and properly documented ?
- [x] Does the figma design fit the ticket description ?
- [x] Added video describing your plan of action ?

## Description

This is part of the base issue [Settings View](https://github.com/omnivore-app/omnivore/issues/193)

Omnivore allows users to create addresses that receive email and add it to a user's inbox. A user can create multiple email addresses. There is a temporary version of this page available at settings/emails.

image

A new design for this page has been created here:

https://www.figma.com/file/ILgs6aXrPfukXOMf5Yibra/Omnivore---Deliverables?node-id=444%3A33319

Notes
The design for this view is slightly off. Visually it is correct, but the user experience is inaccurate. Email addresses are immutable. They can be created, and they can be deleted, but they can not be modified. Also, they are created by Omnivore's backend, so there should be no user input on this page.

The confirmation codes are created when a GMail address is linked with an Omnivore Inbox address. So in testing they will normally not be set.

[ ] List emails

[ ] Display confirmation code if set on an email

[ ] Create an email

[ ] Delete an email

[ ] Copy an email address

[ ] Copy a confirmation code

[ ] Light and dark mode support

[ ] Desktop and mobile support
## Loom
[loom video](https://www.loom.com/share/c70f859f86fd4e93af92da22e0f0a227)

## House Keeping
- [x] Added video describing your plan of action ?
- [x] Added Loom video ?
- [x] Is Deployment Link passing ?
- [x] Have you assinged PR to yourself ?
- [x] Is Github action passing ?
- [x] Updated the appropriate tag ?

## Types of changes

- [ ] Bug fix
- [ ] New feature
- [ ] Refactor
- [x] Others

## Checklist:

- [x] I have test my code on different browsers and devices to make sure it is cross-browser compatible (both desktop and mobile).
- [ ] I have confirmed the payload for integrated API matches params on API docs
- [x] I have confirmed design matches Spec specified on Figma.

## Remarks

- [x] I added a new package to achieve this task.
- [x] I have updated package.json

### CODE REVIEW

**Comment by jacksonh** [comment]
Deploying this one to the demo environment.

**Comment by jacksonh** [comment]
On mobile can we add padding above the table. Its a little too tight to the top of the page

Here's the designed version:

Screen Shot 2022-03-21 at 21 32 59

and here is the actual:

Screen Shot 2022-03-21 at 21 31 41

**Comment by jacksonh** [comment]
On mobile because there is no header, we need to round the top corners of the first item. You can see in our version we have straight lines on the top row on mobile.

**Comment by jacksonh** [comment]
On mobile can we add a little more padding to the two right side buttons. I know that will truncate the address but I think its a little too tight without. 

Screen Shot 2022-03-21 at 21 39 32

**Comment by jacksonh** [comment]
On mobile the bottom button is too tight on the bottom of the page.

Screen Shot 2022-03-21 at 21 42 17

File: packages/web/components/elements/images/MoreOptionsIcon.tsx [hunk 1]
```diff
@@ -7,48 +9,9 @@ type MoreOptionsIconProps = {
 }
 
 export function MoreOptionsIcon(props: MoreOptionsIconProps): JSX.Element {
-  const dots =
```
Comment (jacksonh) [inline packages/web/components/elements/images/MoreOptionsIcon.tsx:10]: Nice 👍

File: packages/web/package.json [hunk 2]
```diff
@@ -23,6 +23,7 @@
     "@radix-ui/react-id": "^0.1.1",
     "@radix-ui/react-popover": "^0.1.1",
     "@radix-ui/react-separator": "^0.1.0",
+    "@radix-ui/react-tooltip": "^0.1.7",
```
Comment (jacksonh) [inline packages/web/package.json:26]: 👍

File: packages/web/pages/_app.tsx [hunk 3]
```diff
@@ -48,7 +49,9 @@ function OmnivoreApp({ Component, pageProps }: AppProps): JSX.Element {
         defaultLocale="en"
         messages={englishTranslations}
       >
-        <Component {...pageProps} />
+        <TooltipProvider delayDuration={200}>
```
Comment (jacksonh) [inline packages/web/pages/_app.tsx:52]: Ah this is handy. Should be able to use these a lot more now.

**Comment by jacksonh** [comment]
On desktop the `InfoIcon` should link to this page when tapped: https://omnivore.app/help/newsletters

**Review by jacksonh — CHANGES_REQUESTED** [review]
added some comments regarding visual issues, mostly on mobile

**Comment by gitstart-omnivore** [comment]
@jacksonh we've effected the requested review changes, please take a look

**Comment by jacksonh** [comment]
> @jacksonh we've effected the requested review changes, please take a look

Thanks, taking a look.

**Comment by jacksonh** [comment]
Deploying this to the demo environment

**Comment by gitstart-omnivore** [comment]
> This should just be the relative URL `/help/newsletters`

Fixed

**Comment by jacksonh** [comment]
This looks good. I'll merge/deploy it in the morning.

**Comment by gitstart-omnivore** [comment]
> There's one last small issue here. On mobile, if you have a single email, you don't get rounded corners, because this loop here doesn't account for the case where `i ===0` and `isLastChild` is true:
> 
> ```
>                   '@mdDown': {
>                     borderRadius: i === 0 ? '5px 5px 0 0 ' : '',
>                   },
>                   borderRadius: isLastChild ? '0 0 5px 5px' : '',
> ```

Fixed

**Comment by jacksonh** [comment]
nice work.

**Review by jacksonh — APPROVED** [review]

**Pull request merged**